### PR TITLE
chore: upstream f5aa05e

### PR DESCRIPTION
**Pull request by elfedy**
# What :computer: 
Include upstream updates until [f5aa05e](https://github.com/foundry-rs/foundry/commit/f5aa05ee9ac4f088db64b93ae5df4d8656faf1ea)

# Why :hand:
Keep up to date with latest foundry

# Evidence :camera:
CI passes with incorporated changes

### CODE REVIEW

**Review by nbaztec — APPROVED** [review]
Relying on CI and previous reviews.

File: crates/cheatcodes/spec/src/vm.rs [hunk 1]
```diff
@@ -1531,6 +1531,14 @@ interface Vm {
...
     function writeLine(string calldata path, string calldata data) external;
 
+    /// Gets the artifact path from code (aka. creation code).
+    #[cheatcode(group = Filesystem)]
+    function getArtifactPathByCode(bytes calldata code) external view returns (string memory path);
+
+    /// Gets the artifact path from deployed code (aka. runtime code).
+    #[cheatcode(group = Filesystem)]
+    function getArtifactPathByDeployedCode(bytes calldata deployedCode) external view returns (string memory path);
+
```
Comment (Karrq) [inline crates/cheatcodes/spec/src/vm.rs:1672]: Should we be supporting this somehow/document that we don't?

**Review by Karrq — APPROVED** [review]
Went commit-by-commit, trying to find possibly interesting features (see previous comment), other than that did a cursory review of your commits, looks good to me.

I'm so glad we have CI to test everything up so far

File: crates/cheatcodes/spec/src/vm.rs (hunk 1 above)
Comment (elfedy) [reply crates/cheatcodes/spec/src/vm.rs:1672]: Probably, I'll do a new feature review after this and open the corresponding issues

**Pull request merged**